### DISCUSSION TITLE
.io ccTLD

### DISCUSSION THREAD

**Discussion by ladd-a** [rust-lang/crates.io#9634] [General]
Since `.io` domain is a country code top level domain (ccTLD) and the country it belongs to is disappearing ([ref. article](https://theconversation.com/uk-hands-chagos-islands-to-mauritius-marking-the-end-of-a-longstanding-sovereignty-dispute-240575)), does crates.io/cargo have a contingency backup available in case the primary TLD ceases operation?

Is this even worth bringing up? It strikes me odd no-one is talking about it, or at least I was not able to find any discussions or issues on it.

**Comment by Turbo87** [comment]
there has been some discussion on Zulip about this. from my understanding we should have at least five years to react to the situation if the domain should actually get deprecated, but so far it's not clear whether that is actually gonna happen at all.

the mitigation would be to use a different domain and build a fallback into cargo. that means older cargo releases potentially would not work anymore, but there is not much we could do about that AFAIU. but again, let's see if this actually happens...

**Comment by ladd-a** [comment]
> that means older cargo releases potentially would not work anymore   

That is my concern; if the fallback is delayed too much and the domain actually goes away, there might not be enough time for all users to upgrade. Of course it should not effect cargo upgrades (since its different domain afaik), so theoretically it would just force everyone to upgrade on deadline date if they wish to continue using official default repositories...

**Reply by YgorSouza** [reply]
I think you'd still be able to use an old version of cargo by setting the new crates.io URL as a mirror as explained in <https://doc.rust-lang.org/cargo/reference/source-replacement.html>

**Comment by purplesyringa** [comment]
For what it's worth, crates.rs exists and is owned by someone from the Rust community. If a switch is necessary, this might be a viable option.

**Reply by morsley** [reply]
I don't think swapping from one country code to another is a good idea should this happen again.  If you don't operate from that country, then you probably shouldn't buy domains there.

**Reply by teohhanhui** [reply]
> If you don't operate from that country, then you probably shouldn't buy domains there.

FWIW: https://www.reddit.com/r/rust/comments/1bl3lr1/serbian_registry_of_internet_domain_names_loves/

**Reply by NULLx76** [reply]
And https://cratesio.com as well: https://github.com/rust-lang/simpleinfra/blob/master/terraform/dns/cratesio.com.tf

**Comment by Alonely0** [comment]
If `.io` does indeed get removed, the most straightforward—yet cursed—thing we can do is host it under `crates.io.rust-lang.org`; alternatively, we could rebrand *crates.io* to just *crates* or *crates.rs* (which is rather problematic), resulting in something more sane like `crates.rust-lang.org` or taking over `crates.rs`. Either way, it's better to do it earlier rather than later, and once a course of action has been set, we ought to make the `crates.io` domain redirect to wherever we end up deciding; we don't really want to start moving domains just months before IANA kills `.io`, so even if it's in a gray area right now, taking action (or at least deciding what should be done) is preferable.

Edit: just saw we also got `cratesio.com`, that's also worth looking at.

**Reply by ladd-a** [reply]
Clinging to the `.io` branding in any way after the TLD goes away seems silly.
I can see the marketing benefit it offers, but something like `crates.rust-lang.org` seems logically the best choice from the user perspective. Of course this is highly subjective matter.

**Comment by oli-obk** [comment]
"Obviously" the right answer is `crat.es` (not linking as it's taken by a domain registrar reseller, and the price goes up the more ppl open the url)
